hummingbot/connector/exchange/bittrex/bittrex_api_user_stream_data_source.py

```diff
@@ -1,160 +1,112 @@
 import asyncio
-import hashlib
-import hmac
-import logging
-import time
-import uuid
-
 from base64 import b64decode
-from typing import Any, AsyncIterable, Dict, List, Optional
-from zlib import decompress, MAX_WBITS
+from typing import Any, AsyncIterable, Dict, Optional
+from zlib import MAX_WBITS, decompress
 
 import signalr_aio
 import ujson
-
 from async_timeout import timeout
 
+from hummingbot.connector.exchange.bittrex import bittrex_constants as CONSTANTS
 from hummingbot.connector.exchange.bittrex.bittrex_auth import BittrexAuth
 from hummingbot.core.data_type.user_stream_tracker_data_source import UserStreamTrackerDataSource
-from hummingbot.logger import HummingbotLogger
-
-BITTREX_WS_FEED = "https://socket-v3.bittrex.com/signalr"
-MAX_RETRIES = 20
-MESSAGE_TIMEOUT = 30.0
-NaN = float("nan")
+from hummingbot.core.web_assistant.ws_assistant import WSAssistant
 
 
 class BittrexAPIUserStreamDataSource(UserStreamTrackerDataSource):
 
-    MESSAGE_TIMEOUT = 30.0
-    PING_TIMEOUT = 10.0
-
-    _btausds_logger: Optional[HummingbotLogger] = None
-
-    @classmethod
-    def logger(cls) -> HummingbotLogger:
-        if cls._btausds_logger is None:
-            cls._btausds_logger = logging.getLogger(__name__)
-        return cls._btausds_logger
-
-    def __init__(self, bittrex_auth: BittrexAuth, trading_pairs: Optional[List[str]] = []):
-        self._bittrex_auth: BittrexAuth = bittrex_auth
-        self._trading_pairs = trading_pairs
-        self._current_listen_key = None
-        self._listen_for_user_stream_task = None
-        self._last_recv_time: float = 0
-        self._websocket_connection: Optional[signalr_aio.Connection] = None
-        self._hub = None
+    def __init__(self, auth: BittrexAuth,
+                 connector,
+                 api_factory):
         super().__init__()
-
-    @property
-    def hub(self):
-        return self._hub
-
-    @hub.setter
-    def hub(self, value):
-        self._hub = value
+        self._auth = auth
+        self._connector = connector
+        self._api_factory = api_factory
+        self._websocket_connection: Optional[signalr_aio.Connection] = None
+        self.hub = None
 
     @property
     def last_recv_time(self) -> float:
-        return self._last_recv_time
+        """
+        Returns the time of the last received message
+
+        :return: the timestamp of the last received message in seconds
+        """
+        if self._ws_assistant:
+            return self._auth.time_provider.time()
+        return 0
```

Comment:
I don't think this is right. Since Bittrex, uses its own ws client, it means we have to manage the `last_recv_time` ourselves.

Reply:
Please @737Baron take a look at this comment. It has not been addressed yet (and it is 7 days old already)

Reply:
Updated the function according to the reworked implementation of signalr.